core/src/main/scala/org/apache/spark/deploy/history/ApplicationHistoryProvider.scala

```diff
@@ -73,4 +78,34 @@ private[history] abstract class ApplicationHistoryProvider {
   @throws(classOf[SparkException])
   def writeEventLogs(appId: String, attemptId: Option[String], zipStream: ZipOutputStream): Unit
 
+  /**
+   * Has an application attempt completed?
+   *
+   * The default returns the completed state of the history info
+   * @param appId application ID
+   * @param attemptId optional attempt ID
+   * @param applicationHistoryInfo the application being probed.
+   * @throws java.util.NoSuchElementException if there is no current record of the application
+   * @return true if the application is considered to have completed
+   */
+  @throws(classOf[NoSuchElementException])
+  def isCompleted(appId: String,
+    attemptId: Option[String],
```

Comment:
nit: each arg on its own line (including the first one).  here and lots of other places.